### ISSUE TITLE
potential miscomputation in get_reconstruction_error of scVI

### ISSUE THREAD

**Issue by eroell** [scverse/scvi-tools#1854]
When calling `get_reconstruction_error` of the `scVI` model, `compute_reconstruction_error` gets called.

Filling/summing an empty dictionary `log_lkl` starting from line [48](https://github.com/scverse/scvi-tools/blob/9e13c6e6d2b12de3afe5f9ec5bb4b69e19b4be64/scvi/model/base/_log_likelihood.py#L48) looks as follows:

```python
        for key, value in rec_loss_dict.items():
            if key in log_lkl:
                log_lkl[key] += torch.sum(value).item()
            else:
                log_lkl[key] = 0.0
```
I think the correct computation would be

```python
        for key, value in rec_loss_dict.items():
            if key in log_lkl:
                log_lkl[key] += torch.sum(value).item()
            else:
                log_lkl[key] = torch.sum(value).item()
```

My apologies in case its correct & I misunderstand the computation involved.

**Comment by adamgayoso** [comment]
Indeed this looks like a bug, thank you for spotting. It seems like the reconstruction loss would be rescaled incorrectly as it would not consider the loss of the first minibatch.

Would you be able to make a PR with this change asap? Thanks!

**Comment by adamgayoso** [comment]
it would also be great to add a test

**Comment by eroell** [comment]
should such a test evaluate the numeric result `get_reconstruction_error`? it appears to me that generally the result of `model` methods are not further tested [here](https://github.com/scverse/scvi-tools/blob/9e13c6e6d2b12de3afe5f9ec5bb4b69e19b4be64/tests/models/test_models.py#L182)...

if yes to numeric result evaluation, a first thought on a test I could insert there, relying on `get_likelihood_parameters`:

```python
params = model.get_likelihood_parameters(adata, give_mean=True)
px = scvi.distributions._negative_binomial.ZeroInflatedNegativeBinomial(mu=torch.tensor(params['mean']),
                                                            theta=torch.tensor(params['dispersions']),
                                                            zi_logits=torch.logit(torch.tensor(params['dropout'])))
zinb_ll = torch.mean(px.log_prob(torch.tensor(adata.X)).sum(axis=1))

assert np.isclose(rec_error['reconstruction_loss'], zinb_ll)
```
the two quantities `zinb_ll` and `rec_error` do deviate by 1-2 percent when I tested this. I suppose numeric issues? (compared to arbitrarily big deviation before the fix, depending on the batch size)

**Comment by adamgayoso** [comment]
for anyone following this issue in the future, this bug appears to have been in existence since the very first versions of scvi-tools. If comparing two models with the same exact dataset (obs in same order), the ranking of methods would still be correct as this bug basically ignores the first minibatch. In all other cases there is no guarantee the rank ordering of outputs (if comparing methods) is accurate.

**Comment by adamgayoso** [comment]
This was fixed in #1857
